kernel/include/feeds.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/ioccom.h>
+
+//
+// Event feed types
+//
+
+typedef enum osquery_event {
+  OSQUERY_EVENT_MY_EVENT
+} osquery_event_t;
+
+// My event for a sample feed.
+typedef struct my_event {
+  uint32_t my_num;
+  char my_str[64];
+} my_event_data_t;
+
+
+//
+// IOCTL messages
+//
+
+// TODO: Think about 32bit daemon on 64bit kernel (different pointer sizes).
```

Comment:
![](https://wwcdn.weddingwire.com/wedding/270001_275000/273079/thumbnails/400x400_1381445527616-morbidforcats.nopenopenope2f68024010405.jpg)

Don't worry about 32bit osquery daemons. We will never let this happen. ;)